app/src/controllers/modal/actionCreators.js

```diff
@@ -29,3 +29,10 @@ export const hideModalAction = () => ({
 export const confirmModalAction = () => ({
   type: CONFIRM_MODAL,
 });
+
+export const setIsSaveButtonPressedAction = (isSaveButtonPressed) => ({
```

Comment:
Previous solution with local state more preferable in this case, because it is not some kind of global state that should be located in redux and not common feature for all modals. So please move it to the component level.